test/Microsoft.AspNetCore.Routing.Tests/DefaultLinkGeneratorTest.cs

```diff
@@ -830,6 +830,35 @@ public void GetLink_InlineConstraints_CompositeInlineConstraint_Fails()
             Assert.False(canGenerateLink);
         }
 
+        private class UpperCaseParameterTransform : ParameterTransformer
+        {
+            public override string Transform(string value)
+            {
+                return value?.ToUpperInvariant();
+            }
+        }
+
+        [Fact]
+        public void GetLink_ParameterTransformer()
+        {
+            // Arrange
+            var endpoint = EndpointFactory.CreateRouteEndpoint("{controller:upper-case}/{name}");
+
+            var routeOptions = new RouteOptions();
+            routeOptions.ConstraintMap["upper-case"] = typeof(UpperCaseParameterTransform);
+
+            var services = GetBasicServices();
+            services.AddSingleton(typeof(UpperCaseParameterTransform), new UpperCaseParameterTransform());
+
+            var linkGenerator = CreateLinkGenerator(new[] { endpoint }, routeOptions, services);
+
+            // Act
+            var link = linkGenerator.GetLink(new { controller = "Home", name = "Test" });
+
+            // Assert
+            Assert.Equal("/HOME/Test", link);
```

Comment:
I would also expect this to be part of DFA matcher when the parameter actually appears in the route template.

So something like an `id` parameter would run the transformer per-request. Something like an `action` parameter would run statically when building the endpoint, and then be ignored.